tools/old_config_maker/graph_node.py

```diff
@@ -0,0 +1,260 @@
+
+        self.add_item('add_check_box', (4, 0), Button(self.tab, text="Add Metric", command=self.add_check_box))
+
+        # Time interval settings
+        self.add_item('lowerTime_lbl', (0, 2, 2), Label(self.tab, text="Time interval(seconds) Lower:", borderwidth=1))
+
+        self.add_item('lowerTime_chk', (2, 2), Entry(self.tab, width=5))
+
+        self.add_item('upperTime_lbl', (3, 2), Label(self.tab, text="Upper:", borderwidth=1))
+
+        self.add_item('upperTime_chk', (4, 2), Entry(self.tab, width=5))
+
+        if values: self.set_values(values)
+
+    @staticmethod
+    def read_available_metrics():
+        """
+        Default metrics to display, read from GraphNode.init_settings_filename
+
+        Returns
+        -------
+        dict of possible metrics and the metrics data.
+        """
+        # print(file_io.possible_metrics(GraphNode.init_settings_filename))
+
+        for filename in GraphNode.init_settings_filename:
+            try:
+                output = file_io.possible_metrics(filename)
+            except IOError:
+                print("Failed to read file: {}".format(filename))
+
+        if not output:
+            raise IOError("Possible metrics file not found!")
+
+        return output
+
+    def add_item(self, name, loc, obj):
+        if type(obj) is dict:
+            for item in obj.values():
+                item.configure(background="#66AA33")
+        elif not type(obj) is Entry:
+            obj.configure(background="#66AA33")
+
+        self.items[name] = obj
+        self.item_locations[name] = loc
+
+    def add_check_box(self):
+        """
+        Add checkbox/metric to be selected
+        """
+
+        def close_config_window(self, master, entries):
+            """
+            Saves all of the data from the window, tries to create a metric and checkboxes and then closes window
+
+            Parameters
+            ----------
+            master: tkinter window
+                Window to read data from and close
+
+            entries: List of tkinter entries
+                Entry boxes to read data from
+            """
+
+            data = [e.get() for e in entries]
+
+            try:
+                metric = Metric(BooleanVar(), label=data[0], func=data[1], x_stream=data[2], y_stream=data[3],
+                                z_stream=data[4])
+
+                self.check_box_values.update({metric.label: metric})
+                self.items['check_boxes'].update({metric.label: Checkbutton(self.tab, text=metric.label, var=metric.output)})
+                self.items['check_boxes'][metric.label].configure(background="#66AA33")
+
+                self.set_grid()
+            except AssertionError as e:
+                print("Error: {}".format(e))
+
+            master.destroy()
+            master.quit()
+
+        master = Tk()
+        master.title("Add Metric")
+        master.geometry('250x130')
+
+        Label(master, text="Label").grid(row=0)
```

Comment:
Doesn't follow dry. labels could easily be created in a loop here where i indicates the row and text indices the name of the label